src/dspaces-server.c

```diff
@@ -2245,7 +2322,20 @@ static void get_rpc(hg_handle_t handle)
     margo_bulk_free(bulk_handle);
     out.ret = dspaces_SUCCESS;
     out.len = csize;
-    obj_data_free(od);
+    /* If we read the od back from the swap space, 
```

Comment:
I don't clearly understand this question.

Reply:
I think I meant here that if you abstract the swapping, this should happen during the swapping operation. It should be in the expanded critical section, in any case.

Reply:
emmm, what I think here is: we should first deliver the data back to the user, and then do the local ls update.
For sure, I can move the critical section to the line after `read_od()`, and just do a conditional `free()` here, but I think it might delay the response to client.

Reply:
Why not find the object to remove and then remove it from local structures in the critical section, then write to disk after the critical section?

Reply:
ah, this is the `get()`. We need to read the data from the file to the memory add add it back to the local storage. This critical section is for the add back.

This is not the `put()`